backend/api/Repositories/Mayan/MayanDocumentRepository.cs

```diff
@@ -124,6 +124,35 @@ public async Task<ExternalResult<QueryResult<DocumentDetail>>> GetDocumentsListA
             return response;
         }
 
+        public async Task<ExternalResult<DocumentDetail>> GetDocumentAsync(long documentId)
+        {
+            _logger.LogDebug("Retrieving document...");
+
+            string authenticationToken = await _authRepository.GetTokenAsync();
+
+            Uri endpoint = new($"{this._config.BaseUri}/documents/{documentId}/");
+            var response = await GetAsync<DocumentDetail>(endpoint, authenticationToken).ConfigureAwait(true);
+
+            _logger.LogDebug("Finished retrieving document");
+            return response;
+        }
+
+        public async Task<ExternalResult<QueryResult<DocumentMetadata>>> GetDocumentMetadataAsync(long documentId, string ordering = "", int? page = null, int? pageSize = null)
+        {
+            _logger.LogDebug("Retrieving document metadata...");
+
+            string authenticationToken = await _authRepository.GetTokenAsync();
+
+            var queryParams = GenerateQueryParams(ordering, page, pageSize);
+
+            string endpointString = $"{_config.BaseUri}/documents/{documentId}/metadata/";
+            Uri endpoint = new(QueryHelpers.AddQueryString(endpointString, queryParams));
+            var response = await GetAsync<QueryResult<DocumentMetadata>>(endpoint, authenticationToken).ConfigureAwait(true);
```

Comment:
Is there a reason we are using configureAwait true here and not in the delete or create functions in this file?

Reply:
No particular one. Once we add the warning back to the config we should see all the ones where configureAwait should have been set